nixos/modules/tasks/network-interfaces.nix

```diff
@@ -310,6 +310,32 @@ let
 
   };
 
+  vswitchInterfaceOpts = {name, ...}: {
+
+    options = {
+
+      name = mkOption {
+        description = "Name of the interface";
+        example = "eth0";
+        type = types.string;
```

Comment:
Don't use `types.string`, it's been deprecated for a long time, same for the `type` option #66346

Reply:
Changed all types to `types.str` instead of `types.string`. Thank you!